### PR TITLE
Start adding nullability annotations to all public or protected symbols

### PR DESCRIPTION
**Pull request by SUPERCILEX**
> ALL HAIL THE LINT GOD, TOR!!! \*thunderous applause\*

Cough cough, ahem. Did you hear something? 😁

Built on top of #1321 which upgrades lint to a version that suggests adding nullability annotations to all public symbols. Also built on top of #1253 so I don't have to step all over myself.

### CODE REVIEW

**Comment by samtstern** [comment]
@SUPERCILEX I targeted this for `4.1.0` but let me know if that's not right.

**Comment by SUPERCILEX** [comment]
Nope, that seems right to me.

**Comment by samtstern** [comment]
@SUPERCILEX this has ~1trillion merge conflicts.  Still want to try and get this in for `4.1.0` or should we punt it?

**Comment by SUPERCILEX** [comment]
Eh, punt it. I'll get back to this later.

**Comment by SUPERCILEX** [comment]
@samtstern This one should be fine to merge along with #1421 since it doesn't touch the auth module. 👍

TBH, I feel like the auth module has too many internal classes where nullability annotations don't matter for us to enable it. 🤷‍♂️

File: auth/build.gradle.kts [hunk 1]
```diff
@@ -11,6 +11,7 @@ android {
 
     lintOptions {
         disable("UnusedQuantity")
+        disable("UnknownNullness") // TODO fix in future PR
```
Comment (samtstern) [inline auth/build.gradle.kts]: I wonder if there's a way we could disable the nullness check on classes that have `@RestrictTo` but leave it on other `public` classes?

**Comment by samtstern** [comment]
@SUPERCILEX LGTM

File: auth/build.gradle.kts (hunk 1 above)
Comment (SUPERCILEX) [reply auth/build.gradle.kts]: Damn, that's a great idea! I'll file an issue for Tor.

Comment (SUPERCILEX) [reply auth/build.gradle.kts]: https://issuetracker.google.com/issues/113439283

**Pull request merged**